### ISSUE TITLE
Refactor `check_all_stablecoin_stability` and rename it to `elasticity_check`

### ISSUE THREAD

**Issue by balqaasem** [setheum/setheum#298]
```
for currency_id in T::StableCurrencyIds::get() {
let checked_diff = Self::get_peg_price_difference(currency_id);
let amount = Self::get_serping_amount_from_ratio(checked_diff); // this takes checked_diff ratio amount from total_supply as ratio amount.
Self::serp_tes(currency_id, amount), // this also adds amount to serp-tes
```
Ref: Refactor SerpTes in SerpTreasury #299

**Comment by balqaasem** [comment]
`on_serp_tes` now replaces `elasticity_check`

**Comment by balqaasem** [comment]
Closes in #307
